mlflow/langchain/retriever_wrapper.py

```diff
@@ -0,0 +1,127 @@
+    retriever: BaseRetriever = Field(exclude=True)
+
+    class Config:
+        """Configuration for this pydantic object."""
+
+        extra = Extra.forbid
+        arbitrary_types_allowed = True
+
+    @property
+    def input_keys(self) -> List[str]:
+        """Return the input keys.
+        :meta private:
+        """
+        return [self.input_key]
+
+    @property
+    def output_keys(self) -> List[str]:
+        """Return the output keys.
+        :meta private:
+        """
+        return [self.output_key]
+
+    def _get_docs(self, question: str) -> List[Document]:
+        """Get documents from the retriever."""
+        return self.retriever.get_relevant_documents(question)
+
+    def _call(
+        self,
+        inputs: Dict[str, Any],
+        run_manager: Optional[CallbackManagerForChainRun] = None,
+    ) -> Dict[str, Any]:
+        """Run _get_docs on input query.
+        Returns the retrieved documents under the key 'source_documents'.
+        Example:
+        .. code-block:: python
+        chain = MlflowRetrieverEvaluator(retriever=...)
+        res = chain({'query': 'This is my query'})
+        docs = res['source_documents']
+        """
+        question = inputs[self.input_key]
+        docs = self._get_docs(question)
+        list_of_str_page_context = [doc.page_content for doc in docs]
+        return {self.output_key: json.dumps(list_of_str_page_context)}
+
+    async def _aget_docs(self, question: str) -> List[Document]:
+        """Get documents from the retriever."""
+        return await self.retriever.aget_relevant_documents(question)
+
+    async def _acall(
+        self,
+        inputs: Dict[str, Any],
+        run_manager: Optional[AsyncCallbackManagerForChainRun] = None,
+    ) -> Dict[str, Any]:
+        """Run _get_docs on input query.
+        Returns the retrieved documents under the key 'source_documents'.
+        Example:
+        .. code-block:: python
+        chain = MlflowRetrieverEvaluator(retriever=...)
+        res = chain({'query': 'This is my query'})
+        docs = res['source_documents']
+        """
+        question = inputs[self.input_key]
+        docs = await self._aget_docs(question)
+        list_of_str_page_context = [doc.page_content for doc in docs]
+        return {self.output_key: json.dumps(list_of_str_page_context)}
+
+    @property
+    def _chain_type(self) -> str:
+        """Return the chain type."""
+        return "retriever_wrapper"
+
+    @classmethod
+    def load(cls, file: Union[str, Path], **kwargs: Any) -> RetrieverWrapper:
+        """Load a RetrieverWrapper from a file."""
+        # Convert file to Path object.
+        if isinstance(file, str):
+            file_path = Path(file)
+        else:
+            file_path = file
+        # Load from either json or yaml.
+        if file_path.suffix == ".json":
+            with open(file_path) as f:
+                config = json.load(f)
+        elif file_path.suffix == ".yaml":
```

Comment:
```suggestion
        elif file_path.suffix in (".yaml", ".yml"):
```

Any chance that `file_path` ends with `.yml`?

Reply:
Not really, since the file is written by langchain, and it's always ".yaml". More context: https://github.com/mlflow/mlflow/pull/8808#discussion_r1264158000

> This load() function is essentially copied from langchain loading.py. Thus, I would prefer not to edit this part, and to keep it in sync with the langchain code. Ideally, we want to contribute the RetrieverWrapper chain to langchain so that we can reused the chain-loading logic. (like the [PR here](https://github.com/hwchase17/langchain/pull/6602/files#diff-ebd95a64222b51c481e180359c531e25bcb3e96a65e6cf32e51ae53a798b6d5cR404))

Reply:
Better to be safe than sorry here. I like @harupy 's a suggestion. Couldn't langchain start writing .yml tomorrow?